lock.go

```diff
@@ -77,6 +77,14 @@ func (l *Lock) IsExpired() bool {
 	return l.isExpired()
 }
 
+// Expiration returns a channel into which the time will be sent upon expiration.
+func (l *Lock) Expiration() <-chan time.Time {
+	if l == nil {
+		return time.After(0)
+	}
+	return time.After(l.leaseDuration - time.Since(l.lookupTime))
```

Comment:
This definition of Expiration is wrong. 

Several concurrency related events can happen while waiting for `time.After` to run its course. And none of them are covered here.

Reply:
see https://github.com/cirello-io/dynamolock/issues/42#issuecomment-525811318